src/components/apps/admin/referenceGenomes/TableView.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * @author sriram
+ *
+ * A table view for admin Reference Genome Listing
+ *
+ */
+
+import React from "react";
+
+import {
+    useRowSelect,
+    useGlobalFilter,
+    useSortBy,
+    useTable,
+} from "react-table";
+
+import refGenomeConstants from "./constants";
+import TableToolbar from "./Toolbar";
+
+import ids from "./ids";
+
+import { build } from "@cyverse-de/ui-lib";
+
+import Checkbox from "@material-ui/core/Checkbox";
+import MaUTable from "@material-ui/core/Table";
+import Paper from "@material-ui/core/Paper";
+
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+
+const IndeterminateCheckbox = React.forwardRef(
+    ({ indeterminate, ...rest }, ref) => {
+        const defaultRef = React.useRef();
+        const resolvedRef = ref || defaultRef;
+
+        React.useEffect(() => {
+            resolvedRef.current.indeterminate = indeterminate;
+        }, [resolvedRef, indeterminate]);
+
+        return (
+            <>
+                <Checkbox ref={resolvedRef} {...rest} />
+            </>
+        );
+    }
+);
+
+const EnhancedTable = ({
+    baseId,
+    columns,
+    data,
+    onAddClicked,
+    onDeletedClicked,
+}) => {
+    const {
+        getTableProps,
+        headerGroups,
+        prepareRow,
+        rows,
+        preGlobalFilteredRows,
+        setGlobalFilter,
+        //selectedFlatRows,
+        state: { /*selectedRowIds,*/ globalFilter },
+        toggleRowSelected,
+    } = useTable(
+        {
+            columns,
+            data,
+            initialState: {
+                sortBy: [
+                    {
+                        id: refGenomeConstants.keys.NAME,
+                        desc: false,
+                    },
+                ],
+            },
+        },
+        useGlobalFilter,
+        useSortBy,
+        useRowSelect,
```

Comment:
Thats pretty much the config needed for filtering, sorting and row selection to work.